real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LogicErrorMessageConsole.java

```diff
@@ -0,0 +1,173 @@
+				buttons[k] = new JButton(robobuggyMessageLevels[k].toString());
+			}
+			
+			
+			JLabel label = new JLabel("Logic Errors");
+			EXCEPTION_btn = new JButton("SHOW EXCEPTION");
+			EXCEPTION_btn.addActionListener(new ActionListener() {
+				
+				@Override
+				public void actionPerformed(ActionEvent e) {
+					show_messages[0] = !show_messages[0];
+					if(show_messages[0]){
+						EXCEPTION_btn.setBackground(Color.green);
+						EXCEPTION_btn.setText("SHOW EXCEPTION");
+					}else{
+						EXCEPTION_btn.setBackground(Color.gray);
+						EXCEPTION_btn.setText("DONT SHOW EXCEPTION");
+
+					}
+				}
+			});
+			WARNING_btn = new JButton("SHOW WARNING");		
+			WARNING_btn.addActionListener(new ActionListener() {
+				
+				@Override
+				public void actionPerformed(ActionEvent e) {
+					show_messages[1] = !show_messages[1];
+					if(show_messages[1]){
+						WARNING_btn.setBackground(Color.GREEN);
+						WARNING_btn.setText("SHOW WARNING");
+					}else{
+						WARNING_btn.setBackground(Color.GRAY);
+						WARNING_btn.setText("DONT SHOW WARNING");
+					}
+				}
+			});
+			NOTE_btn = new JButton("SHOW NOTE");
+			NOTE_btn.addActionListener(new ActionListener() {
+				
+				@Override
+				public void actionPerformed(ActionEvent e) {
+					show_messages[2] = !show_messages[2];
+					if(show_messages[2]){
+						NOTE_btn.setBackground(Color.GREEN);
+						NOTE_btn.setText("SHOW NOTE");
+					}else{
+						NOTE_btn.setBackground(Color.GRAY);
+						NOTE_btn.setText("DONT SHOW NOTE");
+
+					}
+					
+					
+				}
+			});
+			this.addComponent(label, 0, 0, .25, 1);
+			this.addComponent(EXCEPTION_btn, .25, 0, .25, 1);
+			this.addComponent(WARNING_btn, .50, 0, .25, 1);
+			this.addComponent(NOTE_btn, .75, 0, .25, 1);
+
+
+		}
+
+
+	}
+	public void addMessage(String newMessage){
+		messageBuffer[messageBufferEnd] = newMessage;
+		String text = "";
+		for(int i = 0;i< messageBuffer.length;i++){
+			text+=messageBuffer[(messageBufferStart+i)%messageBuffer.length]+"\n";
+		}
+		messageBufferEnd = (messageBufferEnd +1)%messageBuffer.length;
+		//move start up if need be
+		if(messageBufferEnd == messageBufferStart){
+			messageBufferStart = (messageBufferStart +1)%messageBuffer.length;
+		}
+		messages.setText(text);
+		Gui.getInstance().fixPaint();
+		
+	}
+	
+	
+	public LogicErrorMessageConsole(){
+		Header header = new Header();
+		EXCEPTION_btn.setEnabled(true);
+		WARNING_btn.setEnabled(true);
+		NOTE_btn.setEnabled(true);
+		 messages = new JTextArea();
+		this.addComponent(header, 0.0, 0.0, 1.0, .1);
+		this.addComponent(messages, 0.0, 0.1, 1.0, .9);
+		messages.setText("this is where user defined messages will go\n");
+		
+		// Subscriber for LogicException updates
+		new Subscriber(NodeChannel.LOGIC_EXCEPTION.getMsgPath(), new MessageListener() {
+			@Override
+			public void actionPerformed(String topicName, Message m) {
+				RobobuggyLogicExceptionMeasurment msg = (RobobuggyLogicExceptionMeasurment)m;
+				switch(msg.getLevel()){
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [A switch statement does not contain a break](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1760190883/issues/source?bid=2938665&fileBranchId=3059924#l144)